### PR TITLE
(Re-)Add API endpoints for list manipulations

### LINKED ISSUE #1297: v5 : php script sub.php is missing
- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**
8, I can read/write PHP

---
**Expected behaviour:**

Domains are removed from the whitelist/blacklist when making a POST to admin/api.php?list=black&sub=domain.com
admin/api.php?list=black&add=domain.com works fine

**Actual behaviour:**

Domains are not added to blacklist when making a POST to admin/api.php?list=black&sub=domain.com

**Steps to reproduce:**

do a POST to admin/api.php?list=black&add=domain.com
Check blacklist: domain is added : OK
do a POST to admin/api.php?list=black&sub=domain.com
Check blacklist : domain is still there NOK

**Troubleshooting undertaken, and/or other relevant information:**

Opened api.php to see what it refers to
It refers to add.php and sub.php 
add.php is available but sub.php is missing.

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:** 

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes, and have included unit tests where possible.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

- Fix #1297
- Finally get rid of the remaining `add.php` and use the `groups.php` routines for everything (eases maintenance).

**How does this PR accomplish the above?:**

Add API endpoints `api.php?list=black&add=domain.com` like commands where `add=domain` and `sub=domain` can be used to manage the lists. A simple `api.php?list=black` will simply list all blacklisted domains.

`list` can be `black`, `white`, `regex_white` and `regex_black`.

**What documentation changes (if any) are needed to support this PR?:**

None

### CODE REVIEW

File: scripts/pi-hole/js/queries.js [hunk 1]
```diff
@@ -39,19 +39,22 @@ function add(domain, list) {
...
+        comment: "Added from Query Log"
+      },
       success: function (response) {
         alProcessing.hide();
-        if (
-          response.indexOf("not a valid argument") !== -1 ||
-          response.indexOf("is not a valid domain") !== -1 ||
-          response.indexOf("Wrong token") !== -1
-        ) {
+        console.log(response);
```
Comment (XhmikosR) [inline scripts/pi-hole/js/queries.js]: `console.log` should be removed. I'll add a rule to check for this later.

**Comment by PromoFaux** [comment]
Presumably this PR also overrides #1360?

**Comment by DL6ER** [comment]
@PromoFaux Yes

**Comment by PromoFaux** [comment]
@DL6ER please resolve merge conflicts

**Review by PromoFaux — APPROVED** [review]

**Pull request merged**